### DISCUSSION TITLE
Large volume

### DISCUSSION THREAD

**Discussion by kalmiya** [midgen/uesvon#27] [General]
Hello, 

I found this repository, this looks really promising, so decided to give it a try. I added the SVONVolume to the scene and am experimenting with generating the tree, in the context of a space game. So there' is very little objects but in a large volume, let's say 100.000 x 100.000 x 50.000 units (can probably be less but intentionally going high, to understand the limitations)

Taking a brush value for x/y/z of 15k x 15k x 5k is fine, and so is 20k x 20k x 20k.
![image](https://user-images.githubusercontent.com/1215366/212210241-afd4bb8b-dc6f-49d2-8acc-0c20d84e8303.png)

With some values for the x/y/z of the box-brushshape weird leaf-volume cubes start appearing, e.g. 16k x 15k x 5k...  (or 20k x 20k x 10k  or 30k x 30k x 30k)
![image](https://user-images.githubusercontent.com/1215366/212210302-034d9fbc-0b1b-46a3-bb3c-94a3a8ff88ee.png)

I'm wondering what's going on - especially why 20k x 20k x 20k is fine but 20k x 20k x 10k is not, which is weird (seem it's not related to the total volume)... Any clues?

With regards!

**Comment by midgen** [comment]
Hmm, that's an interesting one. Unfortunately I'm just about to go away for a week so can't look at it immediately, I will do when I return though.

**Reply by kalmiya** [reply]
I'm not in a rush, and appreciate your effort to investigate when you have time. 
thanks :)

Random idea, maybe it's only the visualization.

**Comment by kalmiya** [comment]
Hey there, do you have an estimate on when you'll be able to take a look at this? Thanks!

**Comment by midgen** [comment]
I've had a quick play around and can't repro it at the moment.

Could you try making a test level, i.e. one with only basic cubes in it, and finding a repro? I will try and have a look when I can, but fairly busy at the moment.

**Reply by kalmiya** [reply]
Started with an empty level and couldn't reproduce it either. Exported my level into an empty project to minimize assets, and could repro. Then started to delete even more stuff in batches and at some point it worked correctly...  Same procedure, step-by-step and deleting the planets was the key. 

So... it's the rings of the planets, having collisions enabled... using a 500k^3 area and visualizing makes it pretty clear:
![image](https://user-images.githubusercontent.com/1215366/218204313-a2e76930-39ae-4415-91a5-d19393d95124.png)

Good suggestion to make a test-scene, thanks :)
